packages/docprovider/src/yprovider.ts

```diff
@@ -95,6 +92,44 @@ export class WebSocketProviderWithLocks
     this._isInitialized = false;
     this._onConnectionStatus = this._onConnectionStatus.bind(this);
     this.on('status', this._onConnectionStatus);
+
+    const state = options.state;
+    if (name != '' || color != '#' || state == null) {
+      if (name == '') {
+        name = getAnonymousUserName();
+      }
+      if (color == '#') {
+        color = '#' + getRandomColor().slice(1);
+      }
+      // only set if this was not already set by another plugin
+      if (currState && currState.name == null) {
+        options.ymodel.awareness.setLocalStateField('user', {
+          name,
+          color
+        });
+      }
+      return;
+    }
+
+    const user = state.fetch(USER);
```

Comment:
Now that the `state` object is optional, maybe we should check whether it is null or not. Probably in JupyterLab, this will never be null but could be null in other apps.

Reply:
Okay, never mind. I did not realize you are checking in the line 97.